pkg/kubelet/apis/config/types.go

```diff
@@ -622,6 +622,27 @@ type SerializedNodeConfigSource struct {
 	Source v1.NodeConfigSource
 }
 
+// ServiceAccountTokenCacheType is the type of cache key used for caching credentials returned by the plugin
+// when the service account token is used.
+type ServiceAccountTokenCacheType string
+
+const (
+	// TokenServiceAccountTokenCacheType means the kubelet will cache returned credentials
+	// on a per-pod basis. This should be set if the returned credential's lifetime is limited
+	// to the input serviceAccount+pod+serviceAccountToken's lifetime.
+	// For example, this must be used when returning the input serviceAccountToken directly as a pull credential.
+	TokenServiceAccountTokenCacheType ServiceAccountTokenCacheType = "Token"
+	// PodServiceAccountTokenCacheType means the kubelet will cache returned credentials
+	// on a per-pod basis. This should be set if the returned credential is only valid
+	// for the serviceAccount+pod which triggered the image pull attempt.
+	PodServiceAccountTokenCacheType ServiceAccountTokenCacheType = "Pod"
```

Comment:
after discussion in https://github.com/kubernetes/kubernetes/pull/132771, we couldn't come up with a good scenario for pod-level caching... let's drop this for now and just do ServiceAccount level and Token level